index.js

```diff
@@ -0,0 +1,32 @@
+document.addEventListener("DOMContentLoaded",() => {
+    //let inputValue = document.querySelector("input").value
+    let button = document.querySelector("button")
+
+    button.addEventListener('click',async(e) => {
+        e.preventDefault()
+        let ul = document.querySelector("ul")
+        let li = document.createElement('li')
+        let inputValue = document.querySelector("input")
+        res = await axios.get(`http://localhost:3000/animal/${inputValue.value}`)
```

Comment:
If you want `res` to be a global variable, declare it on the top `let res;` doing it the way you're doing it works but it doesn't look good.